src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/api/FluidVolume.java

```diff
@@ -0,0 +1,223 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api;
+
+import java.util.AbstractCollection;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Objects;
+
+import com.google.common.collect.Iterators;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.properties.FluidPropertyManager;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+/**
+ * a mutable representation of a quantity and it's data of a fluid.
+ * this class can and should be extended for custom logic when dealing with volume merging.
+ *
+ * @see #merge(FluidVolume)
+ * @see #drain(FluidVolume)
+ * @see #draw(Fraction)
+ * @see FixedSizedFluidVolume
+ */
+public class FluidVolume extends AbstractCollection<FluidContainer> implements FluidContainer, SingleFluidContainer {
```

Comment:
If we're gonna have a mutable fluid volume, it should have a way to listen to mutations. I use a class called [`Observable`](https://github.com/CottonMC/UniversalComponents/blob/master/src/main/java/io/github/cottonmc/component/api/Observable.java) in Universal Components which lets you just add a runnable listener.

Reply:
resync()

Reply:
Ahhh, that's what that's for. Missed that getting called.